pkg/oc/clusterup/coreinstall/components/pivot/initialconfig.go

```diff
@@ -0,0 +1,144 @@
+package pivot
+
+import (
+	"io/ioutil"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/origin/pkg/oc/clusterup/componentinstall"
+	"github.com/openshift/origin/pkg/oc/clusterup/docker/dockerhelper"
+)
+
+type Component interface {
+	Name() string
+	Install(dockerClient dockerhelper.Interface) error
+}
+
+type KubeAPIServerContent struct {
+	InstallContext componentinstall.Context
+}
+
+func (k *KubeAPIServerContent) Name() string {
+	return "KubeAPIServerContent"
+}
+
+// this ia list of all the secrets and configmaps we need to create
+const (
+	ns = "openshift-kube-apiserver"
+
+	// saTokenSigningCerts contains certificates corresponding to the valid keys that are and were used to sign SA tokens
+	saTokenSigningCerts = "sa-token-signing-certs"
+	// aggregatorClientCABundle is the ca-bundle to use to verify that the aggregator is proxying to your apiserver
+	aggregatorClientCABundle = "aggregator-client-ca"
+	// aggregatorClientCertKeyPair is the client cert/key pair used by the aggregator when proxying
+	aggregatorClientCertKeyPair = "aggregator-client"
+	// kubeletClientCertKeyPair is the client cert/key used by the kube-apiserver when communicating with the kubelet
+	kubeletClientCertKeyPair = "kubelet-client"
+	// kubeletServingCABundle is the ca-bundle to use to verify connections to the kubelet
+	kubeletServingCABundle = "kubelet-serving-ca"
+	// apiserverServingCertKeyPair is the serving cert/key used by the kube-apiserver to secure its https server
+	apiserverServingCertKeyPair = "serving-cert"
+	// apiserverClientCABundle is the ca-bundle used to identify users from incoming connections to the kube-apiserver
+	apiserverClientCABundle = "client-ca"
+	// etcdClientCertKeyPair is the client cert/key pair used by the kube-apiserver when communicating with etcd
+	etcdClientCertKeyPair = "etcd-client"
+	// etcdServingCABundle is the ca-bundle to use to verify connections to etcd
+	etcdServingCABundle = "etcd-serving-ca"
+)
+
+func (c *KubeAPIServerContent) Install(dockerClient dockerhelper.Interface) error {
```

Comment:
why is this a one-time install? Why don't we reconcile?

If it is a one-time thing, we give up control over these files later? I.e. the admin or another operator will maintain (e.g. rotate) them?

Reply:
this is not operator, in "old" cluster up, this will just run and give you cluster, this is not controller-like process

Reply:
oh, right :)